File: 2365-task-scheduler-ii/2365-task-scheduler-ii.py
class Solution:
    def taskSchedulerII(self, tasks: List[int], space: int) -> int:
        Dict = defaultdict(int)
        time = 1
        for i in tasks:
          d = int(Dict[i])
          time = max(time,d)
          Dict.update({i : time + space + 1}) 
          time += 1
        
        return time-1
          

File: 2365-task-scheduler-ii/2365-task-scheduler-ii.cpp
class Solution {
public:
    long long taskSchedulerII(vector<int>& tasks, int space) {
      long long time = 1;
      unordered_map<int,long long> map1;
      
      for (int i = 0; i < tasks.size(); i++)
      {
        time = max(time,map1[tasks[i]]);
        map1[tasks[i]] = time + space + 1;
        time++;
      }
      return time-1;
    }
};